spec.emu

```diff
@@ -48,7 +48,9 @@ copyright: false
       1. Let _buffer_ be a new empty List.
       1. Repeat,
         1. Let _value_ be ? IteratorStepValue(_iterated_).
-        1. If _value_ is ~done~, return ReturnCompletion(*undefined*).
+        1. If _value_ is ~done~, then
+          1. If the number of elements in _buffer_ &lt; ℝ(_windowSize_), throw a *TypeError* exception.
```

Comment:
```suggestion
          1. If _buffer_ is not empty and the number of elements in _buffer_ &lt; ℝ(_windowSize_), throw a *TypeError* exception.
```

Reply:
Because for an empty iterator you can yield zero windows without skipping any items and without yielding an undersized window, which is not true for undersized-but-nonempty.

Reply:
What do you think about this?

```suggestion
          1. If the number of elements in _buffer_ is in the interval from 1 (inclusive) to ℝ(_windowSize_) (exclusive), throw a *TypeError* exception.
```

Reply:
Eh, I find that harder to read than my suggestion, but either's fine for now.